static/js/views/discovered-addon.js

```diff
@@ -123,6 +129,55 @@ class DiscoveredAddon {
         controlDiv.innerHTML = el;
       });
   }
+  
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+		if(e.target.getAttribute('data-id')) {
+			const license_url = "https://api.mozilla-iot.org:8443/addons/license/" + e.target.getAttribute('data-id');
+
+			var modal = document.getElementById('media-modal');
+			
+			if(modal == null){
+		  	var modal_container = document.createElement('div');
+		  	modal_container.className = "media-modal";
+				modal_container.id = "media-modal";
+			
+		  	var modal_frame = document.createElement('div');
+		  	modal_frame.className = "media-modal-frame";
+		  	modal_frame.innerHTML = '<div class="media-modal-close" id="modal-close-button"></div><div class="media-modal-content"><p id="media-modal-text">Loading...</p></div>';
```

Comment:
The `Loading...` bit should go into the translation file, `static/fluent/en-US/main.ftl`. You can then get it via `fluent.getMessage(<id>)`.

Reply:
Yeah I was wondering about that. I think I might remove the loading alltogether, since it's not really visible it things go well.